src/messagecard.ts

```diff
@@ -9,38 +9,110 @@ export function buildMessageCard(
   repoBranch: string,
   avatar_url: string,
   login: string,
-  author_url: string
+  author_url: string,
+  link: string
 ): string {
-  const card = {
-    '@type': 'MessageCard',
-    '@context': 'https://schema.org/extensions',
-    $schema: 'https://adaptivecards.io/schemas/adaptive-card.json',
-    version: '1.0',
-    summary: messageTitle,
-    themeColor: messageColour,
-    title: messageTitle,
-    sections: [
+  let actions: Record<string, string>[] = []
+  if (link !== '') {
+    actions = [
+      {
+        type: 'Action.OpenUrl',
+        title: 'View',
+        url: link,
+        role: 'button'
+      }
+    ]
+  }
+
+  const title = {
+    type: 'TextBlock',
+    size: 'Large',
+    weight: 'Bolder',
+    text: messageTitle,
+    wrap: true
+  }
+
+  const subtitle = {
+    type: 'ColumnSet',
+    columns: [
       {
-        activityTitle: `[${repoName}](${repoUrl})`,
-        activitySubtitle: `by [${login}](${author_url})`,
-        activityImage: avatar_url,
-        facts: [
+        type: 'Column',
+        width: 'auto',
+        items: [
           {
-            name: 'Run Number',
-            value: runNumber
-          },
+            type: 'Image',
+            url: avatar_url,
+            size: 'Small',
+            style: 'Person',
+            altText: login
+          }
+        ]
+      },
+      {
+        type: 'Column',
+        width: 'stretch',
+        items: [
           {
-            name: 'Run ID',
-            value: runId
+            type: 'TextBlock',
+            text: `[${repoName}](${repoUrl})`,
+            weight: 'Bolder',
+            size: 'Medium',
+            wrap: true
           },
           {
-            name: 'Branch',
-            value: repoBranch
+            type: 'TextBlock',
+            text: `by [${login}](${author_url})`,
+            weight: 'Bolder',
+            size: 'Small',
+            wrap: true
           }
-        ],
-        text: messageBody
+        ]
+      }
+    ]
+  }
+
+  const message = {
+    type: 'TextBlock',
+    text: messageBody,
+    wrap: true
+  }
+
+  const facts = {
+    type: 'FactSet',
+    facts: [
+      {
+        title: 'Run Number',
+        value: runNumber
+      },
+      {
+        title: 'Run ID',
+        value: runId
+      },
+      {
+        title: 'Branch',
+        value: repoBranch
+      }
+    ]
+  }
+
+  const adaptiveCard = {
+    type: 'AdaptiveCard',
+    $schema: 'http://adaptivecards.io/schemas/adaptive-card.json',
+    version: '1.4',
+    body: [title, subtitle, message, facts],
+    actions
```

Comment:
Does this need a key?

Reply:
This is covered under: https://eslint.org/docs/latest/rules/object-shorthand

Reply:
Ahh makes sense!